### PR TITLE
Fix broken links - March

### PR DESCRIPTION
**Pull request by gewarren**
Contributes to #23124.

I'll fix the duplicate alt-text warnings in a separate PR.

### CODE REVIEW

File: docs/architecture/dapr-for-net-developers/dapr-at-20000-feet.md [hunk 1]
```diff
@@ -150,7 +150,6 @@ At the time of this writing, the following component types are provided by Dapr:
 | [Bindings](https://github.com/dapr/components-contrib/tree/master/bindings) | Provides a uniform interface to trigger application events from external systems and invoke external systems with optional data payloads. |
 | [Middleware](https://github.com/dapr/components-contrib/tree/master/middleware) | Allows custom middleware to plug into the request processing pipeline and invoke additional actions on a request or response. |
 | [Secret stores](https://github.com/dapr/components-contrib/tree/master/secretstores) | Provides a uniform interface to interact with external secret stores, including cloud, edge, commercial, open-source services. |
-| [Tracing exporters](https://github.com/dapr/components-contrib/tree/master/exporters) | Provides a uniform interface to open telemetry wrappers. |
```
Comment (gewarren) [inline docs/architecture/dapr-for-net-developers/dapr-at-20000-feet.md:153]: https://github.com/dapr/dapr/issues/2337

File: docs/architecture/microservices/secure-net-microservices-web-applications/azure-key-vault-protects-secrets.md [hunk 2]
```diff
@@ -42,9 +42,6 @@ Note that calling `AddAzureKeyVault` requires the application ID that was regist
 - **Data Protection key management and lifetime in ASP.NET Core** \
   [https://docs.microsoft.com/aspnet/core/security/data-protection/configuration/default-settings](/aspnet/core/security/data-protection/configuration/default-settings)
 
-- **Microsoft.Extensions.Configuration** GitHub repository. \
-  <https://github.com/dotnet/extensions/tree/master/src/Configuration>
```
Comment (gewarren) [inline docs/architecture/microservices/secure-net-microservices-web-applications/azure-key-vault-protects-secrets.md:46]: https://github.com/dotnet/extensions/pull/3833

File: docs/core/migration/20-21.md [hunk 3]
```diff
@@ -19,7 +19,7 @@ For an overview of the new features in .NET Core 2.1, see [What's new in .NET Co
 
 - Remove `<DotNetCliToolReference>` references for tools that are bundled in the .NET Core 2.1 SDK (v 2.1.300 or later). These references include:
 
-  - [dotnet-watch](https://github.com/dotnet/aspnetcore/blob/master/src/Tools/dotnet-watch/README.md) (Microsoft.DotNet.Watcher.Tools)
```
Comment (gewarren) [inline docs/core/migration/20-21.md:22]: https://github.com/dotnet/aspnetcore/pull/29997

**Review by Youssef1313 — COMMENTED** [review]
Would you be able to fix https://github.com/dotnet/docs/issues/23268 as well, and potentially get the tool that discovers broken links updated to detect this?

**Review by IEvangelist — APPROVED** [review]
Looks good to me, :shipit: when you're ready.

**Comment by gewarren** [comment]
> Would you be able to fix #23268 as well, and potentially get the tool that discovers broken links updated to detect this?

I can fix that in a separate PR. Thanks for letting me know.

**Pull request merged**